doc/api/modules.md

```diff
@@ -6,7 +6,9 @@
 
 <!--name=module-->
 
-In the Node.js module system, each file is treated as a separate module. For
+CommonJS modules are the classic format to package JavaScript code for Node.js.
+
```

Comment:
```suggestion
Node.js also supports the cross-platform standard [ECMAScript modules][].

```

I think it’s important to call out and link to the ESM docs here, and it’s good to imply that they’re more a compatible choice. We don’t need to dive into the details of which platforms they’re compatible with and what needs to be done to write cross-platform code, but we also don’t need to avoid all mention that ESM offers more compatibility.

Reply:
I know there's a lot of nuance and history here so maybe I'm misguided on this--and if so, please feel totally free to ignore me--but I'd prefer avoiding "cross-platform" because it has always implied operating systems to me. Maybe "Node.js also supports [ECMAScript modules][] which are also supported in browsers." or something similar to that?

Reply:
ESM is also supported in Deno, and Cloudflare Workers, to name just two non-browser environments. I’d rather not restrict this to “Node and browsers” which implies that Node is the only non-browser platform of significance. It might even be _more_ relevant that people want to write cross-platform server-side code, rather than cross-platform full-stack code.

The structure of the sentence is such that the other platforms that support ESM need to be things that are comparable to Node; the sentence says that Node is one of the supporters, so therefore the other supporters (the other platforms) need to be things that fall in the same category as Node. So even if someone initially thinks of operating systems as being platforms, the sentence structure should clue them in that since Node isn’t an operating system, the “platforms” in question here need to be other JavaScript runtimes like Node.

Reply:
```suggestion
Node.js also supports the standard [ECMAScript modules][], which allows to write more
portable code.

```
```suggestion
Node.js also supports the cross-ecosystem standard [ECMAScript modules][].

```
wdyt?

Reply:
Why not just "the standard" ?

Reply:
"the language standard"? there's far more than one "standard", and CJS is also a standard, just a node-specific one.

Reply:
To be clear, my suggestion was to remove "cross-ecosystem" from the sentence and leave "the standard ECMAScript modules".

Reply:
The cross-platform aspect is the point.

Reply:
I don't think it's necessary to emphasize this point.

Reply:
> I don’t think it’s necessary to emphasize this point.

So you’re proposing “Node.js also supports standard ECMAScript modules.” instead of “Node.js also supports the cross-platform standard ECMAScript modules.”? Personally I find the former to be so uninformative as to be almost suspect. As a reader new to this, I’d read it and be like “. . . and? Why does Node support two systems? Is the lack of any information about ECMAScript modules meant to imply that I shouldn’t use it? Or is there a certain condition under which it’s preferable, and if so, what?”

The primary reason to use ESM is because it’s the cross-platform, cross-runtime, official standard. That’s a big reason to prefer it over CommonJS. We _do_ get people posting issues on here asking us about which system they should use, since Node has been taking an officially neutral stance that both module systems are first-class and fully supported. The answer is that they can use whichever they prefer, and they might prefer ESM if they want their code to be potentially runnable in places other than Node. Saying so in the docs would help alleviate some confusion.

Reply:
runnable in not-node *without a build process*, is the important caveat i think helps - because CJS is always runnable anywhere, with a build process.

Reply:
Tbh, I think this sentence can be added in a separate PR because this PR is about adding an introduction sentence about CJS, not ESM.

Reply:
I'd prefer to use `cross-runtime` over `cross-platform`: cross-platform is a given in Node.js (any JS you write for Node.js would work on all supported platforms except a few documented exceptions). Cross-runtime is what would be interesting for some users, but also I agree with Jordan that it's more of an opinion or a bet on future state of things at this point – and I'd say it's OK to have opinionated docs as long as those opinions have consensus, but that doesn't seem to be the case here.

What do you think on the following (maybe less informative, but more likely to have consensus?)

```markdown
CommonJS modules are the classic format to package JavaScript code for Node.js.
It's not to be confused with standard [ECMAScript modules][].
```

Reply:
“It’s not to be confused” feels weird; was there much confusion between the two? And I think it’s important to mention the compatibility reason that someone might want to look at ESM.

```
CommonJS modules are the original way to package JavaScript code for Node.js.
Node.js also supports the [ECMAScript modules][] standard supported by
browsers and other JavaScript runtimes.
```